### ISSUE TITLE
Internal Server Error

### ISSUE THREAD

**Issue by dbrown199747** [craftcms/cms#7833]
### Description

I am constantly getting an internal server error in the admin center when viewing different pages. I have viewed the console when these errors appear, and it tells me the issue but i'm unsure how to go about fixing them.

Firstly an example of this, is when I go to Utilities > Updates I am getting an internal server error, and in the console it says "Failed to load resource: the server responded with a status of 403 () - /index.php?p=fitcms/actions/app/cache-updates:1"

![Screen Shot 2021-04-26 at 13 43 15](https://user-images.githubusercontent.com/83223160/116084157-61700000-a695-11eb-9070-8b3fe6a8c4cf.png)

Another example, when loading an entry, i'm getting an error that says "POST https://telent.com/index.php?p=fitcms/actions/seo/seo/render-data 403" and below that it says "Uncaught (in promise) TypeError: Cannot read property 'hasOwnProperty' of null
    at e.<anonymous> (Snippet.js:76)
    at i (runtime.js:45)
    at Generator._invoke (runtime.js:271)
    at Generator.e.<computed> [as next] (runtime.js:97)
    at l (asyncToGenerator.js:3)
    at r (asyncToGenerator.js:25)"

![Screen Shot 2021-04-26 at 13 46 18](https://user-images.githubusercontent.com/83223160/116084510-cc213b80-a695-11eb-894b-06d8aae3a997.png)

Do you have any ideas as to why this is happening and how to potentially fix it. Our client has reported this issue to us multiple times and is getting quite frustrated about it. I've tried updating craft in the past but has never seemed to solve the issue.

Many thanks.

### Additional info

- Craft version: 3.5.19.1
- PHP version: 7.3.27
- Database driver & version: MySQL 5.5.5
- Plugins & versions: SEO plugin version - "ether/seo": "3.6.2"

**Comment by brandonkelly** [comment]
403 errors are related to permission enforcement.

Are you seeing this when logged in as an admin? Or just on a client/author account?

**Comment by dbrown199747** [comment]
Thanks for getting back to me.

Now you say that, I am getting a lot of 403 errors being logged in the storage/logs folder. As an example, this one is appearing a lot:

- [error][yii\web\HttpException:404] yii\web\NotFoundHttpException: Template not found: autodiscover/autodiscover.xml in /home/telent/vendor/craftcms/cms/src/controllers/TemplatesController.php:92

I've done multiple composer updates in the past, as I thought this would install any folders I am missing, but if it's to do with permissions instead that would make sense.

This is when I am logged in as an admin.

The permissions of my vendor folder is 755 and folder inside the vendor folder are also 755. Could you advise if this is correct, or what I can do to resolve the error please.

Many thanks.

**Comment by brandonkelly** [comment]
> Now you say that, I am getting a lot of 403 errors being logged in the storage/logs folder. As an example, this one is appearing a lot:
> 
> * [error][yii\web\HttpException:404] yii\web\NotFoundHttpException: Template not found: autodiscover/autodiscover.xml in /home/telent/vendor/craftcms/cms/src/controllers/TemplatesController.php:92

That just means that something is trying to request `/autodiscover/autodiscover.xml` on your site, and there’s no template (or other route) set up to handle it. 404s are generally nothing to worry about, which is why they get logged to their own special `web-404s.log` log file by default.

> I've done multiple composer updates in the past, as I thought this would install any folders I am missing, but if it's to do with permissions instead that would make sense.

Composer commands will only modify the `vendor/` folder; nothing outside of it.

> The permissions of my vendor folder is 755 and folder inside the vendor folder are also 755. Could you advise if this is correct, or what I can do to resolve the error please.

Sorry to be clear, I meant _user_ permissions, not folder permissions. If you’re seeing these errors while logged in as an admin, we can rule that out though, as admins are permitted to do whatever they want.

Can you please send your Composer files (`composer.json` and `composer.lock`) and a database backup over to support@craftcms.com, along with steps to reproduce the 403 errors? We can try to look into what’s going on from there.

**Comment by lukebailey** [comment]
We have just started seeing this same issue across all our environment, specifically when hitting `index.php?p=a6m1n/actions/app/cache-updates` there is a 403 forbidden error when logged in as an admin to the CP. So far we do not see any other issues running on Craft 3.6.15

**Comment by brandonkelly** [comment]
If either of you can send SSH + admin credentials over to support@craftcms.com, we can try to investigate it from there.

**Comment by lukebailey** [comment]
False alarm on our part, AWS WAF has started to block post requests to get updates in the CP - assume this is because it looks like some sort of XSS.

**Comment by ccchapman** [comment]
Yes we also experienced this and narrowed it down to AWS WAF. We allowed it through there and the issue went away. I think POST request body has all of the lock version constraints and it got flagged.
